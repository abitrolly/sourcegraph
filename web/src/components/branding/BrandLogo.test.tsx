import React from 'react'
import { BrandLogo } from './BrandLogo'
import { mount } from 'enzyme'

describe('BrandLogo', () => {
    test('renders light sourcegraph logo', () =>
        expect(mount(<BrandLogo assetsRoot="/assets" isLightTheme={true} />)).toMatchSnapshot())

    test('renders dark sourcegraph logo', () =>
        expect(mount(<BrandLogo assetsRoot="/assets" isLightTheme={false} />)).toMatchSnapshot())

    test('renders light custom branding logo', () =>
        expect(
            mount(
                <BrandLogo
                    branding={{ brandName: 'b', light: { logo: 'l' } }}
                    assetsRoot="/assets"
                    isLightTheme={true}
                />
            )
        ).toMatchSnapshot())

    test('renders dark custom branding logo', () =>
        expect(
            mount(
                <BrandLogo
                    branding={{ brandName: 'b', dark: { logo: 'l' } }}
                    assetsRoot="/assets"
                    isLightTheme={false}
                />
            )
        ).toMatchSnapshot())
})
