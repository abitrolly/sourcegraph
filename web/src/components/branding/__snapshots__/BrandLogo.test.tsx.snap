// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandLogo renders dark custom branding logo 1`] = `
<BrandLogo
  assetsRoot="/assets"
  branding={
    Object {
      "brandName": "b",
      "dark": Object {
        "logo": "l",
      },
    }
  }
  isLightTheme={false}
>
  <img
    src="l"
  />
</BrandLogo>
`;

exports[`BrandLogo renders dark sourcegraph logo 1`] = `
<BrandLogo
  assetsRoot="/assets"
  isLightTheme={false}
>
  <img
    src="/assets/img/sourcegraph-head-logo.svg"
  />
</BrandLogo>
`;

exports[`BrandLogo renders light custom branding logo 1`] = `
<BrandLogo
  assetsRoot="/assets"
  branding={
    Object {
      "brandName": "b",
      "light": Object {
        "logo": "l",
      },
    }
  }
  isLightTheme={true}
>
  <img
    src="l"
  />
</BrandLogo>
`;

exports[`BrandLogo renders light sourcegraph logo 1`] = `
<BrandLogo
  assetsRoot="/assets"
  isLightTheme={true}
>
  <img
    src="/assets/img/sourcegraph-light-head-logo.svg"
  />
</BrandLogo>
`;
